torchx/schedulers/test/ray_scheduler_test.py

```
@@ -321,10 +326,19 @@ def test_list_throws_without_address(self) -> None:
             if "RAY_ADDRESS" in os.environ:
                 del os.environ["RAY_ADDRESS"]
             with self.assertRaisesRegex(
-                Exception, "RAY_ADDRESS env variable is expected"
+                Exception, "RAY_ADDRESS env variable"
             ):
                 self._scheduler.list()
 
+        def test_list_doesnt_throw_with_client(self) -> None:
+            ray_client = JobSubmissionClient(address="https://test.com")
+            ray_client.list_jobs = MagicMock(return_value=[])
+            _scheduler_with_client = RayScheduler("client_session", ray_client)
+            try:
+                _scheduler_with_client.list()
+            except:
+                self.fail("Unexpected when listing jobs.")
```

we can just let it throw and add a comment that we're testing for success